docs/developers/developer-reference/sc-to-sc-calls.md

```diff
@@ -0,0 +1,472 @@
+
+| `sync` calls | `async` calls |
+| ------------ | ------------- |
+| Are executed immediately, inline. | Are executed after the current transaction is completed. |
+| Only work in the same shard. | Work both in-shard and cross-shard. |
+| Function results are available immediately. | Function results are only available later, in the callback, if a callback exists. |
+| A callee crash causes caller to immediately crash as well. | A callee crash does not cause caller to crash, the error can be caught in the callback, if it exists. |
+| Reentrancy issues possible. | Reentrancy not possible. |
+
+
+[comment]: # (mx-context-auto)
+
+### Transfer-execute calls
+
+Transfer-execute calls are basically async calls without callback. You can think of them as a "fire and forget" mechanism.
+
+They come in two flavors:
+- Transfer-only - they are used to move EGLD or ESDT balance, and nothing else. Balance transfers can rarely fail, so they are very convenient to use.
+- Transfer-and-execute - used when the caller does not care what the result of the callee function is.
+
+They were very important in async v1, because they were the only mechanism to have more than one call leaving a transaction.
+
+[comment]: # (mx-context-auto)
+
+### Error recovery
+
+Error recovery is not possible with sync calls, so sometimes contracts might choose to perform async calls to allow themselves to recover from errors, even if the target contract is in the same shard.
+
+This also means that no error handling is necessary (or effective) around sync calls.
+
+[comment]: # (mx-context-auto)
+
+### Reentrancy
+
+Reentrancy is very unlikely on a MultiversX blockchain, but not impossible, so let's dedicate a chapter to it.
+
+There are 2 main elements that make it less of an issue than on other blockchain architectures:
+1. Native tokens are built into the system, so they cannot become malicious. Most reentrancy attacks are performed via malicious tokens that exploit intermediate states in the middle of function execution.
+2. Reentrancy can only be performed in sync calls. Async calls do not interrupt the execution, so they are not vectors for an attack.
+
+Even so, to avoid vulnerabilities it is best to perform sync calls either at the very beginning, or at the very end of a SC endpoint execution.
+
+A similar problem to reentrancy is the management of the intermediate state between an async call and the processing of its callback. Great care must be taken, to ensure the state of the contract is not vulnerable to attacks in this interval.
+
+
+[comment]: # (mx-context-auto)
+
+### Async callbacks
+
+Async calls can optionally register callbacks. The callbacks are called by the system, irrespective of whether the caller completed successfully or failed.
+
+The callback receives the following inputs:
+- The call result, which contains several values:
+    - In case of success:
+        - The status code, 0 in this case, followed by
```

Comment:
it stops suddenly

Reply:
It was meant to continue on the next line. Rephrased for clarity.